Enable optimizedText for CBOR

felixbarny opened the pull request.

felixbarny committed:
Enable optimizedText for CBOR

felixbarny committed:
Fix registration of ESCborFactory

jordan-powers reviewed (commented):
The outline looks good, but I think there are a couple of bugs related to the parser state. We want to avoid disrupting the parser so that it still works as normal even after calling `getValueAsText()`.

felixbarny committed:
More efficient implementation of Text#stringLength

Doesn't need to decode, just counts the number of characters while iterating over the utf8-encoded bytes.

felixbarny committed:
Fix state handling in ESCborParser

felixbarny commented:
Good catch, thanks!

Commit:
[CI] Auto commit changes from spotless

felixbarny committed:
Add appropriate attribution to ESCborParser

felixbarny committed:
Adjust TextTests: calling stringLength() doesn't initialize a string

felixbarny committed:
Allow apache license in x-content lib

felixbarny committed:
Merge branch 'main' into cbor-optimized-text

felixbarny committed:
Remove @Test annotation

felixbarny committed:
Merge remote-tracking branch 'origin/main' into cbor-optimized-text

felixbarny committed:
Merge branch 'main' into cbor-optimized-text

jordan-powers reviewed (approved):
LGTM, thanks Felix!

felixbarny committed:
Merge branch 'main' into cbor-optimized-text
